sw/device/lib/testing/test_framework/check.h

```diff
@@ -101,90 +101,107 @@
       uint64_t: "CHECK-fail: [%d] both equal: 0x%016x")
 // clang-format on
 
+/**
+ * Compare `actual_` against `ref_` buffer.
+ *
+ * Prints matches between `actual_` and `ref_` before logging an error.
+ *
+ * @param expect_eq_ True if the arrays are expected to be equal, false
+ * otherwise.
+ * @param actual_ Buffer containing actual values.
+ * @param ref_ Buffer containing the reference values.
+ * @param size_ Number of items to compare.
+ * @param ... Arguments to a LOG_* macro, which are evaluated if the check.
+ * @return Either `kOk` or `kInternal`.
+ */
+
+#define CHECK_ARRAYS_IMPL(expect_eq_, actual_, ref_, size_, ...)           \
+  ({                                                                       \
+    static_assert(sizeof(*(actual_)) == sizeof(*(ref_)),                   \
+                  "CHECK_ARRAYS requires arguments of equal size.");       \
+    status_t sts_ = OK_STATUS();                                           \
+    volatile bool is_eq =                                                  \
+        memcmp((actual_), (ref_), size_ * sizeof(*(actual_))) == 0;        \
```

Comment:
The comparison is now implemented with `memcmp` rather than a loop over all elements. What is the rationale for this?

Reply:
The earlier implementation used `memcmp` for `CHECK_ARRAY_NE` and a loop for `CHECK_ARRAY_EQ`, this PR merge the implementations based on using `memcmp`.